### PR TITLE
Improve line info construction

### PR DESCRIPTION
**Pull request by zherczeg**
- Simplify small encoding
- Better line info for some corner cases

### CODE REVIEW

File: tests/unit-core/test-backtrace.c [hunk 1]
```diff
@@ -159,14 +159,14 @@ class_backtrace_callback (jerry_backtrace_frame_t *frame_p, /* frame information
   {
     TEST_ASSERT (jerry_backtrace_is_strict (frame_p));
     TEST_ASSERT (location_p->line == 3);
-    TEST_ASSERT (location_p->column == 12);
+    TEST_ASSERT (location_p->column == 14);
```
Comment (galpeter) [inline tests/unit-core/test-backtrace.c:162]: Why did this changed? Was  the column calculation incorrect before?

Comment (zherczeg) [reply tests/unit-core/test-backtrace.c:162]: Yes, it was pointed to the equal sign. Does not makes any sense.

Comment (galpeter) [reply tests/unit-core/test-backtrace.c:162]: ok, thx.

**Review by galpeter — APPROVED** [review]
lgtm

**Review by ossy-szeged — APPROVED** [review]
LGTM
